### ISSUE TITLE
default partition() crashes on unicode

### ISSUE THREAD

**Issue by rspeer** [discoproject/disco#265]
The default partition() function runs hash(str(key)), which means it can partition anything but Unicode strings.

A simple fix would be to replace this with hash(repr(key)).

**Comment by JensRantil** [comment]
An example test case in case anyone doubt this:

```
$ ipython
In [1]: hash(str(u'å'))
---------------------------------------------------------------------------
UnicodeEncodeError                        Traceback (most recent call last)
/Users/jens/<ipython-input-1-c3acbaef199f> in <module>()
----> 1 hash(str(u'å'))

UnicodeEncodeError: 'ascii' codec can't encode character u'\xe5' in position 0: ordinal not in range(128)
```

**Comment by pmundkur** [comment]
Should be fixed in 0.4.3.
